src/test/java/net/sf/jabref/importer/fileformat/MedlineImporterTest.java

```diff
@@ -0,0 +1,132 @@
+
+/**
+ * Articles in the medline format can be downloaded from http://www.ncbi.nlm.nih.gov/pubmed/.
+ * 1. Search for a term and make sure you have selected the PubMed database
+ * 2. Select the results you want to export by checking their checkboxes
+ * 3. Press on the 'Send to' drop down menu on top of the search results
+ * 4. Select 'File' as Destination and 'XML' as Format
+ * 5. Press 'Create File' to download your search results in a medline xml file
+ *
+ * @author Daniel Mair/Bruehl
+ *
+ */
+public class MedlineImporterTest {
+
+    private final InputStream emptyFileStream = new ByteArrayInputStream("".getBytes(StandardCharsets.UTF_8));
+    private MedlineImporter medlineImporter;
+
+
+    @Before
+    public void setUp() throws Exception {
+        Globals.prefs = JabRefPreferences.getInstance();
+        this.medlineImporter = new MedlineImporter();
+    }
+
+    @Test
+    public void testIsRecognizedFormatAccept() throws Throwable {
+        List<String> list = Arrays.asList("MedlineImporterMinimalEntryTest.xml", "MedlineImporterSingleEntryTest.xml",
+                "MedlineImporterArticleIDTest.xml");
+        for (String str : list) {
+            try (InputStream is = MedlineImporterTest.class.getResourceAsStream(str);) {
+                Assert.assertTrue(medlineImporter.isRecognizedFormat(is));
+            }
+        }
+    }
+
+    @Test
+    public void testIsRecognizedFormatReject() throws Throwable {
+        List<String> list = Arrays.asList("CopacImporterTest1.txt", "CopacImporterTest2.txt", "IEEEImport1.txt",
+                "IsiImporterTestWOS.isi", "oai2.xml", "oai22.xml", "RisImporterTest1.ris");
+        for (String str : list) {
+            try (InputStream is = MedlineImporterTest.class.getResourceAsStream(str);) {
+                Assert.assertFalse(medlineImporter.isRecognizedFormat(is));
+            }
+        }
+    }
+
+    @Test
+    public void testIsRecognizedFormatEmptyFile() throws IOException {
+        Assert.assertFalse(medlineImporter.isRecognizedFormat(emptyFileStream));
+    }
+
+    @Test
+    public void testImportMalformedEntry() throws IOException {
+        try (InputStream is = MedlineImporter.class.getResourceAsStream("MedlineImporterMalformedEntryTest.xml")) {
+            medlineImporter.importEntries(is, new OutputPrinterToNull());
+        }
+    }
+
+    @Test
+    public void testImportEntryWithArticleID() throws IOException {
+        checkImportEntriesAgainstBibFileEntries("MedlineImporterArticleIDTest.xml",
+                "MedlineImporterArticleIDTestBib.bib");
+        try (InputStream shouldBeIs = MedlineImporter.class.getResourceAsStream("MedlineImporterArticleIDTest.xml");
+                InputStream bibIn = MedlineImporter.class.getResourceAsStream("MedlineImporterArticleIDTestBib.bib")) {
+            List<BibEntry> entries = medlineImporter.importEntries(shouldBeIs, new OutputPrinterToNull());
+            Assert.assertEquals(1, entries.size());
```

Comment:
this assertion is not necessary, it is done in BibtexEntryAssert(InputStream, List<BibEntry>)